src/uct/ib/base/ib_iface.h

```diff
@@ -195,6 +198,8 @@ struct uct_ib_iface {
     uint16_t                  pkey_value;
     uint8_t                   addr_size;
     uct_ib_device_gid_info_t  gid_info;
+    unsigned                  seed;
+    unsigned                  rand_value;
```

Comment:
don't need to save both seed and random value
save only random state

Reply:
@yosefe We need saving seed to use it in the next rand_r() calling. We need to save rand_value to make sure the rand_value for the 2nd path is the same as the 1st path. How we can use random state to replace them?

Reply:
random state is actually the random value. so we can do 
```
if (path_index==0) {
  rand_r(&rand_value)
}
udp_sport = rand_value + ...
```

Reply:
@yosefe Fixed